content/en/blog/2022/2022-03-15-k8s-triage-bot-helper-ci-job/K8s Triage bot Helper CI Job_ ‘make update’.md

```diff
@@ -0,0 +1,129 @@
+
+
+### System info:
+
+| OS name | Version | System type |Memory| WSL |
+| ------- | ------- | ----------- | ---  | --- |
+|Windows 11 home|10.0.22000 Build 22000|x64-based|8.00 GB|Ubuntu-20.04 (v2)|
+
+You might encounter the following on your PR
+
+
+The tests failing the particular issue:
+
+![Failed Test Cases](failing_test_cases.png)
+
+Consecutively, 
+
+![Failed Test Cases robot comment](k8s_ci_failed_tests.png)
+
+Additionally one can check the failed test via the `link` provided under details in the above image.
+
+### Potential solution
+Run the failing `.sh` scripts individually known from the CI job output, to generate the expected files to fix up the failures. The `.sh` scripts can be found residing under the `hack/` directory at the root of the `kubernetes/kubernetes` code base. 
+
+ `kubernetes/kubernetes &rarr; hack/ &rarr; .sh update files`
+
+
+
+In this particular case, these files were to be run:
+
+* [hack/update-generated-protobuf.sh](https://github.com/SubhasmitaSw/kubernetes/blob/master/hack/update-generated-protobuf.sh) 
+* [hack/update-generated-swagger-docs.sh](https://github.com/SubhasmitaSw/kubernetes/blob/master/hack/update-generated-swagger-docs.sh)
+* [hack/update-openapi-spec.sh](https://github.com/SubhasmitaSw/kubernetes/blob/master/hack/update-openapi-spec.sh)
+
+After you run these, you might see either of the below issues:
+
+---
+Using Codespaces:
+
+![codespaces error output](codespaces.png)
+![codespaces error output](codespaces2.png)
+
+---
+
+Using Visual Studio Code:
+
+![vscode error output](vscode.png)
+
+### Potential solution to the above error:
+1. Remove the Makefile and Makefile.generated_files files:
+
+```
+rm Makefile Makefile.generated_files
+```
+
+
+2. Create the symlinks:
+
+```
+ln -s build/root/Makefile Makefile
+```
+
+```
+ln -s build/root/Makefile.generated_files Makefile.generated_files
+```
+
+> PS: takes adequate time to generate the scripts on windows.
+
+## The current situation
+
+Many contributors do not have access to powerful environments in which to run `make update` or `make verify`. They can utilise *vscode/wsl/codespaces* and other tools to recommend modifications, but they might get tripped up by `make verify` scripts because in many cases, we end up with files that needs to be re-generated. It's a tall order for them to scan the build log from `make verify` to determine which specific scripts in `hack/` directory they need to run.
+
+### Solution
+
+Two solutions are proposed by [@dims](https://github.com/dims) for the time being:
+
+* **Short Term:** add a CI job that folks can trigger when they need it, that runs `make update` and provides a zip archive that they can be told to download. The archive includes all the files that changed as a result of running `make update`.
+*  **Long Term:** add a new Prow bot command that generates an additional commit on their PR.
```

Comment:
You could make the text “Short Term” and “Long Term” be hyperlinks to the relevant headings.